static/js/pm_list.js

```diff
@@ -147,15 +225,149 @@ export function is_all_privates() {
 }
 
 export function expand() {
-    private_messages_open = true;
+    // Set the localstorage variable of `private_messages_collapsed` as `false`.
+    if (localstorage.supported()) {
+        ls.set("private_messages_collapsed", false);
+    }
+    private_messages_collapsed = false;
     stream_popover.hide_topic_popover();
     update_private_messages();
     if (is_all_privates()) {
-        $(".top_left_private_messages").addClass("active-filter");
+        $("#private_messages_container").addClass("active-filter");
     }
 }
 
 export function update_dom_with_unread_counts(counts) {
     update_private_messages();
     set_count(counts.private_message_count);
 }
+
+export function only_active_pm_out_of_collapsed_pms() {
+    // check for the presence of only active pm outside the list of the collapsed pms.
+    if (
+        $(".expanded-private-messages").children().length === 1 &&
+        $(".expanded-private-messages li.active-sub-filter").is(
+            $(".expanded-private-messages").children()[0],
+        )
+    ) {
+        return true;
+    }
+    return false;
+}
+
+export function append_new_active_pm_in_collapsed_pms() {
+    // We check for the new activated PM thread in our DOM and append it inside the list of pms visible out of
+    // collapsed pms.
+    const convos = _get_convos();
+    const active_convo = convos.find((convo) => {
+        if (convo.is_active === true) {
+            return convo;
+        }
+        return false;
+    });
+    if (active_convo) {
+        // If we find a the new active PM to be present in out convos list then we make a node of it
+        // through keyed_pm_li and update the DOM with vdom.update() function.
+        const node = [pm_list_dom.keyed_pm_li(active_convo)];
+        const new_dom = pm_list_dom.pm_ul(node);
+        const container = ui.get_content_element($("#private_messages_container"));
+        function replace_content(html) {
+            container.html(html);
+        }
+
+        function find() {
+            return container.find("ul");
+        }
+        vdom.update(replace_content, find, new_dom, prior_dom);
+        prior_dom = new_dom;
+    } else {
+        // If we do not find any other active pm in our convos then we clear the list of PMs and
+        // collapse the section.
+        $(".expanded-private-messages").empty();
+        close();
+    }
+}
+
+export function activate_all_private_messages_view() {
+    $("#private_messages_section").addClass("active_private_messages_section");
+    $(".more-private-messages-sidebar-title").css("font-weight", "bold");
```

Comment:
Don't set CSS this way -- set a semantic CSS class, which can happen to use bold as its currently styling.

Reply:
OK, would change this :)